migrations/2_apm.js

```diff
@@ -1,12 +1,14 @@
 const namehash = require('eth-ens-namehash').hash
 const keccak256 = require('js-sha3').keccak_256
 
-const getContract = name => artifacts.require(name)
+module.exports = async (deployer, network, accounts, arts = null) => {
+  if (arts != null) artifacts = arts // allow running outside
 
-const bases = ['APMRegistry', 'Repo', 'ENSSubdomainRegistrar']
-const baseContracts = bases.map(getContract)
+  const getContract = name => artifacts.require(name)
+
+  const bases = ['APMRegistry', 'Repo', 'ENSSubdomainRegistrar']
+  const baseContracts = bases.map(getContract)
```

Comment:
Assuming `artifacts.require()` only takes in one argument, this could just be `bases.map(artifacts.require)`, which might be clearer :).

Reply:
i think artifacts.require takes more than one param